third_party/maya/lib/usdMaya/util.cpp

```diff
@@ -1095,17 +1095,25 @@ _IsShape(const MDagPath& dagPath) {
     return (numberOfShapesDirectlyBelow == 1);
 }
 
-SdfPath
-PxrUsdMayaUtil::MDagPathToUsdPath(const MDagPath& dagPath, bool mergeTransformAndShape)
+std::string
+PxrUsdMayaUtil::MDagPathToUsdPathString(const MDagPath& dagPath)
 {
     std::string usdPathStr(dagPath.fullPathName().asChar());
     std::replace( usdPathStr.begin(), usdPathStr.end(), '|', '/');
-
     // We may want to have another option that allows us to drop namespace's
     // instead of making them part of the path.
     std::replace( usdPathStr.begin(), usdPathStr.end(), ':', '_'); // replace namespace ":" with "_"
+    // Replacing characters because of underworld (transform|shapeNode->|underWorldNode)
+    // Because maya inserts "->|", we can safely eliminate both the "-" and ">"
```

Comment:
No, we want to get rid of -> and keep the rest.

I cleaned up the implementation a bit, first doing the removal then the replaces, kept the iterators and added a bit of extra comment to clarify it further. What do you think, is it easier to follow?
`
std::string usdPathStr(dagPath.fullPathName().asChar());
// We are keeping the iterators around, to avoid extra memory allocations.
// Replacing characters because of underworld (transform|shapeNode->|underWorldNode).
// Maya inserts "->|", we can safely eliminate both the "-" and ">", and keep the rest.
auto itBegin = usdPathStr.begin();
auto itEnd = std::remove(itBegin, usdPathStr.end(), '-');
itEnd = std::remove(itBegin, itEnd, '>');
// Replacing MDagPath separators with the USD ones.
std::replace(itBegin, itEnd, '|', '/');
// We may want to have another option that allows us to drop namespace's
// instead of making them part of the path.
std::replace(itBegin, itEnd, ':', '_');
return std::string(itBegin, itEnd);
`